urld58fd5e7a9_longrm_weibo_robot_tgz-pJ8-weibo4j_org_json_TestJ8/src/weibo4j/Comment.java

```diff
@@ -0,0 +1,318 @@
+     * @since Weibo4J 1.2.0
+     */
+    public long getInReplyToUserId() {
+        return inReplyToUserId;
+    }
+
+    /**
+     * Returns the in_reply_to_screen_name
+     *
+     * @return the in_in_reply_to_screen_name
+     * @since Weibo4J 1.2.0
+     */
+    public String getInReplyToScreenName() {
+        return inReplyToScreenName;
+    }
+
+    /**
+     * returns The location's latitude that this tweet refers to.
+     *
+     * @since Weibo4J 1.2.0
+     */
+    public double getLatitude() {
+        return latitude;
+    }
+
+    /**
+     * returns The location's longitude that this tweet refers to.
+     *
+     * @since Weibo4J 1.2.0
+     */
+    public double getLongitude() {
+        return longitude;
+    }
+
+    /**
+     * Test if the status is favorited
+     *
+     * @return true if favorited
+     * @since Weibo4J 1.2.0
+     */
+    public boolean isFavorited() {
+        return isFavorited;
+    }
+
+
+    private User user = null;
+
+    /**
+     * Return the user
+     *
+     * @return the user
+     */
+    public User getUser() {
+        return user;
+    }
+
+    /**
+     *
+     * @since Weibo4J 1.2.0
+     */
+    public boolean isRetweet(){
+        return null != retweetDetails;
+    }
+
+    /**
+     *
+     * @since Weibo4J 1.2.0
+     */
+    public RetweetDetails getRetweetDetails() {
+        return retweetDetails;
+    }
+
+
+    /*package*/
+    static List<Comment> constructComments(Response res,Weibo weibo) throws WeiboException {
+        Document doc = res.asDocument();
+        if (isRootNodeNilClasses(doc)) {
+            return new ArrayList<Comment>(0);
+        } else {
+            try {
+                ensureRootNodeNameIs("comments", doc);
+                NodeList list = doc.getDocumentElement().getElementsByTagName(
+                        "comment");
+                int size = list.getLength();
+                List<Comment> statuses = new ArrayList<Comment>(size);
+                for (int i = 0; i < size; i++) {
+                    Element status = (Element) list.item(i);
+                    statuses.add(new Comment(res, status, weibo));
+                }
+                return statuses;
+            } catch (WeiboException te) {
+                ensureRootNodeNameIs("nil-classes", doc);
+                return new ArrayList<Comment>(0);
+            }
+        }
+    }
+
+    /* modify by hezhou add some field*/
+    static List<Comment> constructComments(Response res) throws WeiboException {
+   	 try {
+            JSONArray list = res.asJSONArray();
+            int size = list.length();
+            List<Comment> comments = new ArrayList<Comment>(size);
+            for (int i = 0; i < size; i++) {
+            	comments.add(new Comment(list.getJSONObject(i)));
+            }
+            return comments;
+        } catch (JSONException jsone) {
+            throw new WeiboException(jsone);
+        } catch (WeiboException te) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception is caught and re-thrown within the catch clause without any other operations being performed on the exception, making the catch clause unnecessary.

**Fix**: Perform additional operations on the exception that has been caught. You can wrap a standard exception into a custom exception to add additional information or special handling. When you do that, make sure to set the original exception as the cause. Otherwise, you will lose the stack trace and message of the original exception, which will make it difficult to analyze the event that caused the exception.  
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)